examples/governance/tests/RiscZeroGovernorTest.sol

```diff
@@ -0,0 +1,332 @@
+
+        riscZeroGovernor.verifyAndFinalizeVotes(receipt.seal, journal);
+        (
+            uint256 againstVotes,
+            uint256 forVotes,
+            uint256 abstainVotes
+        ) = riscZeroGovernor.proposalVotes(proposalId);
+
+        assertEq(forVotes, 100, "For votes should be 100");
+        assertEq(againstVotes, 50, "Against votes should be 50");
+        assertEq(abstainVotes, 0, "Abstain votes should be 0");
+
+        vm.clearMockedCalls();
+    }
+
+    function testQuorumAndExecution() public {
+        (
+            address[] memory targets,
+            uint256[] memory values,
+            bytes[] memory calldatas,
+            string memory description
+        ) = _createProposalParams();
+
+        uint256 proposalId = riscZeroGovernor.propose(
+            targets,
+            values,
+            calldatas,
+            description
+        );
+
+        aliceSupport = 1;
+        bobSupport = 1;
+
+        vm.roll(block.number + riscZeroGovernor.votingDelay() + 1);
+
+        // Cast votes
+        vm.prank(alice);
+        riscZeroGovernor.castVote(proposalId, 1);
+        vm.prank(bob);
+        riscZeroGovernor.castVote(proposalId, 1);
+
+        vm.roll(block.number + riscZeroGovernor.votingPeriod() + 1);
+
+        (
+            bytes32 finalBallotBoxAccum,
+            bytes memory encodedBallots
+        ) = hashBallots(aliceSupport, bobSupport);
+
+        bytes memory journal = abi.encodePacked(
+            proposalId,
+            finalBallotBoxAccum,
+            encodedBallots
+        );
+        bytes32 journalDigest = sha256(journal);
+
+        VerifierReceipt memory receipt = mockVerifier.mockProve(
+            ImageID.FINALIZE_VOTES_ID,
+            journalDigest
+        );
+
+        // Mock the verifier call
+        address verifierAddress = riscZeroGovernor.verifier.address;
+        bytes4 verifySelector = IRiscZeroVerifier.verify.selector;
+        bytes memory expectedCalldata = abi.encodeWithSelector(
+            verifySelector,
+            receipt.seal,
+            IMAGE_ID,
+            journalDigest
+        );
+        vm.mockCall(verifierAddress, expectedCalldata, abi.encode());
+
+        // Call verifyAndFinalizeVotes
+        riscZeroGovernor.verifyAndFinalizeVotes(receipt.seal, journal);
+
+        assertEq(
+            uint256(riscZeroGovernor.state(proposalId)),
+            uint256(IGovernor.ProposalState.Succeeded),
+            "Proposal should have succeeded"
+        );
+
+        riscZeroGovernor.execute(
+            targets,
+            values,
+            calldatas,
+            keccak256(bytes(description))
+        );
+        assertEq(
+            uint256(riscZeroGovernor.state(proposalId)),
+            uint256(IGovernor.ProposalState.Executed),
+            "Proposal should be executed"
+        );
+
+        vm.clearMockedCalls();
+    }
+
+    function testFailToReachQuorum() public {
```

Comment:
Because `riscZeroGovernor.verifyAndFinalizeVotes` is not called, this test would fail even if enough votes where cast.

Reply:
ah, I thought I had finished this test, but forgot about foundry's `testFail` syntax. Thanks Victor! I've written a new test called `testQuorumNotReached`